### PR TITLE
Logic added to change UK election thrasher depending on user

### PR DESCRIPTION
**Pull request by Mullefa**
cc @guardian/contributions 

__NB__: _do not merge until tested in code_

## What does this change?

Bootstraps the AB testing framework to update the UK Election thrasher depending on whether the reader is either a supporter or contributor, or neither of these.

## What is the value of this and can you measure success?

Allows current supporters and contributors to be thanked; provides another channel for readers who aren't supporters or who haven't contributed to do so.

## Does this affect other platforms - Amp, Apps, etc?

No

## Screenshots

_Coming soon..._

## Tested in CODE?

No

### CODE REVIEW

File: static/src/javascripts-legacy/projects/common/modules/experiments/tests/acquisitions-thrasher-uk-election.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    function onThrasherViewed(callback) {
+        // Safe selecting from array - function will only be called if length of array element is positive.
+        // Element in view logic taken from contribution utilities.
+        elementInView($ukElectionThrasher[0], window, { top: 18 }).on('firstview', callback())
+    }
+
+    return function() {
+
+            this.id = 'AcquisitionsThrasherUkElection';
+
```
Comment (jranks123) [inline static/src/javascripts-legacy/projects/common/modules/experiments/tests/acquisitions-thrasher-uk-election.js]: Is there no campaignId?

File: static/src/javascripts-legacy/projects/common/modules/experiments/tests/acquisitions-thrasher-uk-election.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            this.start = '2017-06-02';
+            this.expiry = '2017-07-03';
+
```
Comment (jranks123) [inline static/src/javascripts-legacy/projects/common/modules/experiments/tests/acquisitions-thrasher-uk-election.js]: the expiry date here and in the ABTestSwitches file  are different

**Review by jranks123 — APPROVED** [review]
Approved (with comments)

File: static/src/javascripts-legacy/projects/common/modules/experiments/tests/acquisitions-thrasher-uk-election.js (hunk 1 above)
Comment (Mullefa) [inline static/src/javascripts-legacy/projects/common/modules/experiments/tests/acquisitions-thrasher-uk-election.js]: Correct - this field is introduced by contribution utilities which I'm not using.

Comment (jranks123) [inline static/src/javascripts-legacy/projects/common/modules/experiments/tests/acquisitions-thrasher-uk-election.js]: 👍

**Review by joelochlann — APPROVED** [review]
Great stuff 👍

**Pull request merged**